devices/sprut.js

```diff
@@ -0,0 +1,47 @@
+const exposes = require('../lib/exposes');
+const fz = require('../converters/fromZigbee');
+const tz = require('../converters/toZigbee');
+const reporting = require('../lib/reporting');
+const e = exposes.presets;
+
+module.exports = [
+    {
+        zigbeeModel: ['WBMSW3'],
+        model: 'WBMSW3',
+        vendor: 'Sprut.device',
+        description: 'WB-MSW v.3 Zigbee Sensor',
+        fromZigbee: [fz.temperature, fz.illuminance, fz.humidity, fz.occupancy, fz.sprut_occupancy, fz.co2, fz.sprut_voc,
+            fz.sprut_noise, fz.sprut_noise_detected, fz.on_off],
+        toZigbee: [tz.on_off],
+        exposes: [e.temperature().withEndpoint('default'), e.illuminance(), e.illuminance_lux(), e.humidity(),
```

Comment:
Why is there an endpoint for temperature? Instead of `e.temperature().withEndpoint('default')` I would expect `e.temperature()`

Reply:
Otherwise it does not work, since the sensor returns the end point in the parameter:
```bash
Zigbee2MQTT:info  2021-12-17 21:04:46: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe8b184e', payload '{"co2":943,"humidity":37.45,"illuminance":9759,"illuminance_lux":9,"last_seen":1639775086756,"linkquality":182,"noise":"38.28","noise_detected":false,"occupancy":false,"occupancy_level":83,"state_l1":"OFF","state_l2":"OFF","state_relay":"OFF","temperature_default":24.31}'
```

Reply:
Should be fixed now, can you confirm?

Reply:
Thanks! Now the temperature has no end point:
```
Zigbee2MQTT:info  2021-12-18 10:06:22: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe8b184e', payload '{"illuminance":11093,"illuminance_lux":13,"last_seen":1639821982780,"linkquality":182,"occupancy_level":77,"state_l1":"OFF","state_l2":"OFF","state_relay":"OFF","temperature":24.17}'
```

It remains to deal with the error when reconnecting. For some reason, it asks for an attribute that was not subscribed to:
```
Zigbee2MQTT:error 2021-12-18 10:07:10: Failed to read state of '0x842e14fffe8b184e' after reconnect (Read 0x842e14fffe8b184e/1 genOnOff(["onOff"], {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE'))
```

Reply:
What if you change `'default': 1` to `'default': 2`?

Reply:
> What if you change `'default': 1` to `'default': 2`?
Replaced, but after reconnection still error
```js
 {
        zigbeeModel: ['WBMSW3'],
        model: 'WBMSW3',
        vendor: 'Sprut.device',
        description: 'WB-MSW v.3 Zigbee Sensor',
        fromZigbee: [fzLocal.temperature, fz.illuminance, fz.humidity, fz.occupancy, fzLocal.occupancy, fz.co2, fzLocal.voc,
            fzLocal.noise, fzLocal.noise_detected, fz.on_off],
        toZigbee: [tz.on_off],
        exposes: [e.temperature(), e.illuminance(), e.illuminance_lux(), e.humidity(),
            e.occupancy(), e.occupancy_level(), e.co2(), e.voc(), e.noise(), e.noise_detected(), e.switch().withEndpoint('default'),
            e.switch().withEndpoint('l2'), e.switch().withEndpoint('relay')],
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint1 = device.getEndpoint(1);
            const binds = ['genBasic', 'msTemperatureMeasurement', 'msIlluminanceMeasurement', 'msRelativeHumidity',
                'msOccupancySensing', 'msCO2', 'sprutVoc', 'sprutNoise'];
            await reporting.bind(endpoint1, coordinatorEndpoint, binds);

            // led_red
            await device.getEndpoint(2).read('genOnOff', ['onOff']);

            // led_green
            await device.getEndpoint(3).read('genOnOff', ['onOff']);

            // buzzer
            await device.getEndpoint(4).read('genOnOff', ['onOff']);

            // Read data at start
            await endpoint1.read('msTemperatureMeasurement', ['measuredValue']);
            await endpoint1.read('msIlluminanceMeasurement', ['measuredValue']);
            await endpoint1.read('msRelativeHumidity', ['measuredValue']);
            await endpoint1.read('msOccupancySensing', ['occupancy']);
            await endpoint1.read('sprutNoise', ['noise']);
            await endpoint1.read('sprutNoise', ['noise_detected']);
        },
        endpoint: (device) => {
            return {'default': 2, 'system': 1, 'l2': 3, 'relay': 4};
        },
        meta: {multiEndpoint: true},
    },
```

```bash
Zigbee2MQTT:warn  2021-12-18 13:19:39: Device '0x842e14fffe8b184e' left the network
Zigbee2MQTT:info  2021-12-18 13:19:39: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x842e14fffe8b184e","ieee_address":"0x842e14fffe8b184e"},"type":"device_leave"}'
Zigbee2MQTT:info  2021-12-18 13:19:39: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"left_network","meta":{"friendly_name":"0x842e14fffe8b184e"},"type":"device_removed"}'
Zigbee2MQTT:debug 2021-12-18 13:19:44: Received MQTT message on 'zigbee2mqtt/bridge/request/permit_join' with data '{"device":null,"time":254,"transaction":"dzdl8-3","value":true}'
Zigbee2MQTT:info  2021-12-18 13:19:44: Zigbee: allowing new devices to join.
Zigbee2MQTT:info  2021-12-18 13:19:44: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"time":254,"value":true},"status":"ok","transaction":"dzdl8-3"}'
Zigbee2MQTT:debug 2021-12-18 13:19:44: Received MQTT message on 'zigbee2mqtt/bridge/request/permit_join' with data 'true'
Zigbee2MQTT:info  2021-12-18 13:19:44: Zigbee: allowing new devices to join.
Zigbee2MQTT:info  2021-12-18 13:19:44: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"value":true},"status":"ok"}'
Zigbee2MQTT:info  2021-12-18 13:19:50: Device '0x842e14fffe8b184e' joined
Zigbee2MQTT:info  2021-12-18 13:19:50: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x842e14fffe8b184e","ieee_address":"0x842e14fffe8b184e"},"type":"device_joined"}'
Zigbee2MQTT:info  2021-12-18 13:19:50: Configuring '0x842e14fffe8b184e'
Zigbee2MQTT:info  2021-12-18 13:19:50: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"0x842e14fffe8b184e"},"type":"device_connected"}'
Zigbee2MQTT:info  2021-12-18 13:19:50: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe8b184e', payload '{"last_seen":1639833590817,"linkquality":182}'
Zigbee2MQTT:info  2021-12-18 13:19:51: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe8b184e', payload '{"last_seen":1639833591180,"linkquality":182}'
Zigbee2MQTT:debug 2021-12-18 13:19:51: Device '0x842e14fffe8b184e' announced itself
Zigbee2MQTT:info  2021-12-18 13:19:51: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x842e14fffe8b184e","ieee_address":"0x842e14fffe8b184e"},"type":"device_announce"}'
Zigbee2MQTT:info  2021-12-18 13:19:51: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x842e14fffe8b184e"},"type":"device_announced"}'
Zigbee2MQTT:debug 2021-12-18 13:19:53: Retrieving state of '0x842e14fffe8b184e' after reconnect
Zigbee2MQTT:debug 2021-12-18 13:19:53: Received Zigbee message from '0x842e14fffe8b184e', type 'readResponse', cluster 'genOnOff', data '{"onOff":0}' from endpoint 2 with groupID 0
Zigbee2MQTT:info  2021-12-18 13:19:53: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe8b184e', payload '{"last_seen":1639833593470,"linkquality":187,"state_default":"OFF"}'
Zigbee2MQTT:debug 2021-12-18 13:20:00: Received MQTT message on 'zigbee2mqtt/0x842e14fffe8b184e/l2/set' with data '{"state_l2":"ON"}'
Zigbee2MQTT:debug 2021-12-18 13:20:00: Publishing 'set' 'state' to '0x842e14fffe8b184e'
Zigbee2MQTT:info  2021-12-18 13:20:00: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe8b184e', payload '{"last_seen":1639833600290,"linkquality":184,"state_default":"OFF"}'
Zigbee2MQTT:info  2021-12-18 13:20:00: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe8b184e', payload '{"last_seen":1639833600290,"linkquality":184,"state_default":"OFF","state_l2":"ON"}'
Zigbee2MQTT:error 2021-12-18 13:20:01: Failed to configure '0x842e14fffe8b184e', attempt 1 (Error: Bind 0x842e14fffe8b184e/1 genBasic from '0x00124b00209d0142/1' failed (AREQ - ZDO - bindRsp after 10000ms)
    at Timeout._onTimeout (/mnt/data/root/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7))
Zigbee2MQTT:debug 2021-12-18 13:20:06: Received MQTT message on 'zigbee2mqtt/0x842e14fffe8b184e/l2/set' with data '{"state_l2":"OFF"}'
Zigbee2MQTT:debug 2021-12-18 13:20:06: Publishing 'set' 'state' to '0x842e14fffe8b184e'
Zigbee2MQTT:info  2021-12-18 13:20:06: Configuring '0x842e14fffe8b184e'

```

Reply:
> What if you change `'default': 1` to `'default': 2`?

Replaced and access to unknown attribute disappeared. Thanks! I'll send in the changes now.